gli/utils.py

```diff
@@ -392,3 +392,366 @@ def save_data(prefix, **kwargs):
     for key, matrix in sparse_arrays.items():
         sp.save_npz(f"{prefix}_{key}.sparse.npz", matrix)
         print("Save sparse matrix", key, "to", f"{prefix}_{key}.sparse.npz")
+
+class HjsonTemplate(object):
+    def __init__(
+        self,
+        name
+    ):
+        self.missing_features_messages_error = []
+        self.missing_features_messages_warning = []
+        self._name = name
+
+    def _check_file_and_key(self, d, v, k):
+        if "file" not in d:
+            self._missing_features_messages_warning.append("%s missing 'file' in %s" % (self._name, v))
+            d.update({"file" : "%s_task.npz" % self._name})
+        if "key" not in d:
+            self._missing_features_messages_warning.append("%s missing 'key' in %s" % (self._name, v))
+            d.update({"key" : k})
+        return d
+    
+    def metadata(
+        self,
+        node=dict(),
+        edge=dict(),
+        graph=dict(),
+        citation=None,
+        is_heterogeneous=True
```

Comment:
We might want to make the interface even easier for users. For example, for homogeneous graph, we may want the function to take in these parameters:

```python
def graph(self,
    edges,
    node_list,
    edge_list=None,
    node_attrs={},
    edge_attrs={},
    graph_attrs={},
    citation=None,
)
```
The first two inputs are corresponding to our required properties. The `edge_list` is an optional property. And all other inputs are optional.

Reply:
Also, should we split the saving interface of homo- and hetero-graph. What do
you think? @jiaqima

Reply:
Yeah, I think it's better to split the interface for these two, which may help significantly simplify the API

Reply:
Add a class called `Attribute` with member `description`, `data`, `type`, `format`.